src/datacatalogtordf/agent.py

```diff
@@ -0,0 +1,148 @@
+"""Agent module for mapping a agent to rdf.
+
+This module contains methods for mapping a agent object to rdf
+according to the
+`dcat-ap-no v.2 standard <https://doc.difi.no/review/dcat-ap-no/#klasse-distribusjon>`__
+
+Example:
+    >>> from datacatalogtordf import Agent, Dataset
+    >>> dataset = Dataset()
+    >>> dataset.identifier = "http://example.com/datasets/1"
+    >>> # Create an agent:
+    >>> agent = Agent()
+    >>> agent.identifier = "http://example.com/agents/1"
+    >>> agent.name = {"en": "James Bond", "nb": "Djeims Bånd"}
+    >>> # Assigen the agent to the publisher property:
+    >>> dataset.publisher = agent
+    >>> bool(dataset.to_rdf())
+    True
+"""
+from __future__ import annotations
+
+from typing import Optional
+
+from rdflib import BNode, Graph, Literal, Namespace, RDF, URIRef
+
+from .uri import URI
+
+
+DCT = Namespace("http://purl.org/dc/terms/")
+DCAT = Namespace("http://www.w3.org/ns/dcat#")
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+
+
+class Agent:
+    """A class representing a foaf:Agent.
+
+    Attributes:
+        identifier (URI): A URI uniquely identifying the agent
+        name (dict): A name given to the agent. key is langauge code
+        organization_id (str): The organzation's identifier
+        type (URI): a link to a concept designating the type of the agent
+    """
+
+    slots = ("_identifier", "_name", "_organization_id")
+
+    _identifier: URI
+    _name: dict
+    _organization_id: str
+    _type: URI
+
+    def __init__(self) -> None:
+        """Inits an object with default values."""
+        # set up graph and namespaces:
+        self._g = Graph()
+        self._g.bind("dct", DCT)
+        self._g.bind("dcat", DCAT)
+        self._g.bind("foaf", FOAF)
+
+    @property
+    def identifier(self: Agent) -> str:
+        """Get/set for identifier."""
+        return self._identifier
+
+    @identifier.setter
+    def identifier(self: Agent, identifier: str) -> None:
+        self._identifier = URI(identifier)
+
+    @property
+    def name(self: Agent) -> dict:
+        """Name attribute."""
+        return self._name
+
+    @name.setter
+    def name(self: Agent, name: dict) -> None:
+        self._name = name
+
+    @property
+    def organization_id(self: Agent) -> str:
+        """Organization attribute."""
+        return self._organization_id
+
+    @organization_id.setter
+    def organization_id(self: Agent, organization_id: str) -> None:
+        self._organization_id = organization_id
+
+    @property
+    def type(self: Agent) -> str:
+        """Type attribute."""
+        return self._type
+
+    @type.setter
+    def type(self: Agent, type: str) -> None:
```

Comment:
Ikke nødvendig med setter for type, ikke sant?

Reply:
Jo, for det er snakk om egenskapen "organization-type", som eg forkorta til "type". Kanskje litt tvetydig. Det er denne det er snakk om: https://data.norge.no/specification/dcat-ap-no/#Akt%C3%B8r-utgivertype

Eg renamer til "organization_type". Godt spotta!